src/Text/Layout/Table/Cell/MultiFormatted.hs

```diff
@@ -0,0 +1,90 @@
+{-# LANGUAGE DeriveTraversable #-}
+
+-- | Provides formatting to an instance of 'Cell'. For example, in a unix
+-- terminal one could use the following:
+--
+-- >>> buildCell (formatted "\ESC[31m" "Hello World!" "\ESC[0m") :: String
+-- Hello World!
+--
+-- The text then appears in dull red.
+module Text.Layout.Table.Cell.MultiFormatted
+    ( MultiFormatted
+    , multiFormatted
+    , formatted
+    , plain
+    ) where
+
+import Data.List (foldl', mapAccumL, mapAccumR)
+import Data.String
+
+import Text.Layout.Table.Primitives.AlignInfo
+import Text.Layout.Table.Cell
+import Text.Layout.Table.StringBuilder
+import Text.Layout.Table.StringBuilder
+
+data MultiFormatted a
+    = MultiFormatted
+      { prefix      :: String
+      , contentList :: [MultiFormatted a]
+      , suffix      :: String
+      }
+    | PlainContent a
+  deriving (Eq, Show, Functor, Foldable, Traversable)
+
+-- | Create a value from content that is kept plain without any formatting.
+plain :: a -> MultiFormatted a
+plain = PlainContent
+
+-- | Create a formatted value with formatting directives that are applied to
+-- the whole value. The actual formatting has to be done by the backend.
+multiFormatted
+    :: String -- ^ Prefix text directives for formatting.
+    -> [MultiFormatted a] -- ^ The content to be formatted.
+    -> String -- ^ Suffix text directives for formatting.
+    -> MultiFormatted a
+multiFormatted = MultiFormatted
+
+-- | Create a formatted value with formatting directives that are applied to
+-- the whole value. The actual formatting has to be done by the backend.
+formatted
+    :: String -- ^ Prefix text directives for formatting.
+    -> a -- ^ The content to be formatted.
+    -> String -- ^ Suffix text directives for formatting.
+    -> MultiFormatted a
+formatted p x s = MultiFormatted p [PlainContent x] s
+
+instance IsString a => IsString (MultiFormatted a) where
+    fromString = plain . fromString
+
+-- The Semigroup instance is not fully lawful, as it is not associative.
+-- However, it is associative after projecting through the Cell class methods.
+instance Semigroup (MultiFormatted a) where
+    a <> b = MultiFormatted "" [a, b] ""
+
+-- The Monoid instance is not fully lawful, as mempty is not a left or right identity.
+-- However, it is an identity after projecting through the Cell class methods.
+instance Monoid (MultiFormatted a) where
+    mempty = MultiFormatted "" [] ""
+
+instance Cell a => Cell (MultiFormatted a) where
+    dropLeft  i = snd . mapAccumL (dropTrackRemaining dropLeft) i
+    dropRight i = snd . mapAccumR (dropTrackRemaining dropRight) i
+    visibleLength = sum . fmap visibleLength
+    measureAlignment p = foldl' (mergeAlign p) mempty
+    buildCell = buildCellMF
+
+-- | Drop either from the right or left, tracking how much extra there is to drop.
```

Comment:
Is the comma correct here?

Reply:
For me it seems more natural to have a comma, but there's a [bit of flexibility on the topic](https://en.wikipedia.org/wiki/Dependent_clause#English_punctuation). I'll rephrase it so that it's more obvious which is correct.